src/core/macros.systemd.in

```diff
@@ -41,25 +41,25 @@ OrderWithRequires(postun): systemd \
 %{nil}
 
 %systemd_post() \
-if [ $1 -eq 1 ] ; then \
+if [ $1 -eq 1 ] && [ -x @bindir@/systemctl ] ; then \
         # Initial installation \
-        systemctl --no-reload preset %{?*} >/dev/null 2>&1 || : \
+        @bindir@/systemctl --no-reload preset %{?*} || : \
```

Comment:
I think this is OK, somewhat surprisingly. `preset` operates without complaint in chroot, and handles the case where the unit doesn't have an [Install] section too.

Reply:
Because there is a `[ -x @bindir@/systemctl ]` added beforehand, which tests that specific path. (Just on the unlikely off chance that an unqualified "systemctl" resolves to a path other than bindir.)

Reply:
OK. I generally prefer a non-absolute path, mainly for didactic reasons: rpm has a legacy of cargo-culting absolute paths everywhere, with stuff like `%_mkdir` and and `%_make` and other obfuscations. But for `%_systemctl` it doesn't really matter, and indeed it would be weird to check for `/usr/bin/systemctl` and then use `systemctl` right below.

Reply:
Avoid `@bindir@` is possible, but it's far from beautiful..
```
x=$(which systemctl 2>/dev/null); # new dependency on /usr/bin/which, yuck
if [ -n "$x" ]; then "$x" preset ...; fi
```

Reply:
Yes, I agree `which` is not worth the trouble.